packages/airnode-node/src/evm/requests/blocking.ts

```diff
@@ -46,6 +45,41 @@ export function blockRequestsWithWithdrawals(requests: GroupedRequests): LogsDat
     return { ...acc, apiCalls: [...acc.apiCalls, apiCall] };
   }, initialState);
 
-  const updatedRequests = { ...requests, apiCalls };
-  return [logs, updatedRequests];
+  return [[...prevLogs, ...logs], { ...requests, apiCalls }];
+}
+
+export function applySponsorRequestLimit([prevLogs, requests]: LogsData<GroupedRequests>): LogsData<GroupedRequests> {
+  const requestCountPerSponsor = new Map<string, number>();
+  const allowedApiCalls: Request<ApiCall>[] = [];
+  const logs: PendingLog[] = [];
+
+  requests.apiCalls.forEach((apiCall) => {
+    if (apiCall.status !== RequestStatus.Pending) return;
+
+    if (!requestCountPerSponsor.has(apiCall.sponsorAddress)) {
+      requestCountPerSponsor.set(apiCall.sponsorAddress, 0);
+    }
+    const sponsorRequests = requestCountPerSponsor.get(apiCall.sponsorAddress)!;
+
+    if (sponsorRequests < MAXIMUM_SPONSOR_WALLET_REQUESTS) {
+      requestCountPerSponsor.set(apiCall.sponsorAddress, sponsorRequests + 1);
+      allowedApiCalls.push(apiCall);
+    } else {
+      logs.push(logger.pend('WARN', `Ignoring Request ID:${apiCall.id} as it exceeded sponsor wallet request limit.`));
+      const blockedCall: Request<ApiCall> = {
+        ...apiCall,
+        status: RequestStatus.Ignored,
```

Comment:
See comment below about ignored vs blocked. I think this needs to be changed
```suggestion
        status: RequestStatus.Blocked,
```